m4/ax_mpi.m4

```diff
@@ -139,12 +139,12 @@ dnl We have to use AC_TRY_COMPILE and not AC_CHECK_HEADER because the
 dnl latter uses $CPP, not $CC (which may be mpicc).
 AC_LANG_CASE([C], [if test x != x"$MPILIBS"; then
 	AC_MSG_CHECKING([for mpi.h])
-	AC_TRY_COMPILE([#include <mpi.h>],[],[AC_MSG_RESULT(yes)], [MPILIBS=""
+	AC_COMPILE_IFELSE([AC_LANG_PROGRAM([],[include <mpi.h>])],[AC_MSG_RESULT(yes)], [MPILIBS=""
 		AC_MSG_RESULT(no)])
 fi],
 [C++], [if test x != x"$MPILIBS"; then
 	AC_MSG_CHECKING([for mpi.h])
-	AC_TRY_COMPILE([#include <mpi.h>],[],[AC_MSG_RESULT(yes)], [MPILIBS=""
+	AC_COMPILE_IFELSE([AC_LANG_PROGRAM([],[include <mpi.h>])],[AC_MSG_RESULT(yes)], [MPILIBS=""
```

Comment:
Should there be a `#` before `include` here @TApplencourt ?

Reply:
Oh yeah indeed.  `https://github.com/emacs-ng/emacs-ng/blob/d90af7b852caa1956c584d7c85ecfff605749e0b/m4/acl.m4#L62` if emacs is pyting `#` I guess we need it!